crypto/BRCryptoWalletManager.c

```diff
@@ -66,18 +86,43 @@ static void
 cwmGetBlockNumberAsBTC (BRWalletManagerClientContext context,
                         BRWalletManager manager,
                         int rid) {
-    BRCryptoWalletManager cwm = context;
-    cwm->client.btc.funcGetBlockNumber (cwm->client.context, manager, rid);
+    BRCryptoWalletManager cwm = cryptoWalletManagerTake (context);
+
+    BRCryptoCWMCallbackHandle record = calloc (1, sizeof(struct BRCryptoCWMCallbackRecord));
+    record->type = CWM_CALLBACK_TYPE_BTC_GET_BLOCK_NUMBER;
+    record->rid = rid;
+
+    cwm->client.btc.funcGetBlockNumber (cwm->client.context, cwm, record);
+
+    cryptoWalletManagerGive (cwm);
 }
 
 static void
 cwmGetTransactionsAsBTC (BRWalletManagerClientContext context,
-                         BRWalletManager btcManager,
+                         BRWalletManager manager,
                          uint64_t begBlockNumber,
                          uint64_t endBlockNumber,
                          int rid) {
-    BRCryptoWalletManager cwm = context;
-    cwm->client.btc.funcGetTransactions (cwm->client.context, btcManager, begBlockNumber, endBlockNumber, rid);
+    BRCryptoWalletManager cwm = cryptoWalletManagerTake (context);
```

Comment:
This function is the tricky one since the Java (and Swift) currently "loop", announcing transactions, discovering addresses, repeatedly.

I think that behaviour (the looping nature of the get transaction calls) needs to be in BRWalletManager.